chore(writing-a-loader): added data sharing details (#3526)
webpack/webpack.js.org: src/content/contribute/writing-a-loader.md

T> The fact that loaders can be chained also means they don't necessarily have to output JavaScript. As long as the next loader in the chain can handle its output, the loader can return any type of module.

### Data sharing

> As discussed earlier, webpack loaders can be chained. So, along with passing the content(source code) with the next chained loader, we can even share data to next loader. There are two ways to share data among loaders.

Reviewer: Can you please point me at the second way of sharing the data?
```suggestion
webpack loaders can be chained and share data with the next chained loader. To share data with the next chained loader, pass it alongside the content (source code). Do it by using the `this.callback` method of raw loaders. In raw method _(the default exported function)_, pass data using the fourth argument of `this.callback`:
```

> __Using `this.callback` method of raw loaders__

Reviewer: ```suggestion
```

Author: @EugeneHlushko I think this line might be needed as there is one `callback` for pitch methods as well and they wont here pass the data to next loaders

Author: cc @EugeneHlushko , do we need to remove this ?

Reviewer: This paragraph above is speaking about `this.callback` method of raw loaders. So does the bold part here, its duplication. Previously it was claiming two different ways of doing it and enlisting only one. Since we've done the rephrasing, it no more claims to have two ways of doing it, therefore this enlistment isnt required in my opinion. Unless you can add second way of doing it with an example

> 

Reviewer: ```suggestion
```

> In raw method _(the default exported function)_, we can pass data to next chained loader using the fourth argument of `this.callback`

Reviewer: ```suggestion
```

> 

Reviewer: ```suggestion
```

Author: not able to apply this

> Example

Reviewer: ```suggestion
```

T> The fact that loaders can be chained also means they don't necessarily have to output JavaScript. As long as the next loader in the chain can handle its output, the loader can return any type of module.

### Data sharing

> As discussed earlier, webpack loaders can be chained. So, along with passing the content with the next chained loader, we can even share data to next loader. There are two ways to share data among loaders.

Reviewer 2: ```suggestion
As discussed earlier, webpack loaders can be chained. So, along with passing content to the next chained loader, we can even share data with the next loader. There are two ways to share data among loaders.
```

> As discussed earlier, webpack loaders can be chained. So, along with passing content to the next chained loader, we can even share data with the next loader. There are two ways to share data among loaders.

Reviewer 3: > So, along with passing content to the next chained loader, we can even share data with the next loader.

This oration says basically the same thing. It is there an important difference?

Author: actually content here is the source code but we can we share other data as well to next loader using 
`this.callback`

the one with pitch method is not correct as it can only pass data from pitch phase to normal phase of the same loaders.

I will remove the pitch example

- __Through pitch method__

>   Webpack loader has a concept of pitching loaders. You can find a detailed explanation [`here`](https://webpack.js.org/api/loaders/#pitching-loader). Here anything assigned to `data` can be accessed by the previous loader's `this.data`. It is mainly used to share data between the pitch phase and normal phase

Reviewer 2: ```suggestion
  webpack loader has a concept of pitching loaders. You can find a detailed explanation [`here`](https://webpack.js.org/api/loaders/#pitching-loader). Here anything assigned to `data` can be accessed by the previous loader's `this.data`. It is mainly used to share data between the pitch phase and normal phase.
```

>   webpack loader has a concept of pitching loaders. You can find a detailed explanation [`here`](https://webpack.js.org/api/loaders/#pitching-loader). Here anything assigned to `data` can be accessed by the previous loader's `this.data`. It is mainly used to share data between the pitch phase and normal phase.

Reviewer 3: ```suggestionwebpack loader has a concept of pitching loaders. You can find a detailed explanation [`here`](/api/loaders/#pitching-loader). Here anything assigned to `data` can be accessed by the previous loader's `this.data`. It is mainly used to share data between the pitch phase and normal phase.
```

>   For example, in the preview chained loader, if `jade-loader` needs to share some data to `apply-loader`, it can assign the data to `data.x` in its pitch phase and `apply-loader` can access that using `this.data.x`.

Reviewer 3: Why adding the space before the paragraph?

- __Using `this.callback` method of raw loaders__

  In raw method _(the default exported function)_, we can pass data to next chained loader using the fourth argument of `this.callback`

>   example

Reviewer 3: ```suggestion
Example:
```
